runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/WindmillStateReaderTest.java

```diff
@@ -99,6 +113,713 @@ private ByteString intData(int value) throws IOException {
+                            .setEntryName(STATE_MULTIMAP_KEY_1)
+                            .setFetchMaxBytes(WindmillStateReader.CONTINUATION_MAX_MULTIMAP_BYTES)
+                            .setRequestPosition(500)
+                            .build()));
+
+    Windmill.KeyedGetDataResponse.Builder response2 =
+        Windmill.KeyedGetDataResponse.newBuilder()
+            .setKey(DATA_KEY)
+            .addTagMultimaps(
+                Windmill.TagMultimapFetchResponse.newBuilder()
+                    .setTag(STATE_KEY_1)
+                    .setStateFamily(STATE_FAMILY)
+                    .addEntries(
+                        Windmill.TagMultimapEntry.newBuilder()
+                            .setEntryName(STATE_MULTIMAP_KEY_1)
+                            .addAllValues(Arrays.asList(intData(7), intData(8)))
+                            .setContinuationPosition(800)
+                            .setRequestPosition(500)));
+    Windmill.KeyedGetDataRequest.Builder expectedRequest3 =
+        Windmill.KeyedGetDataRequest.newBuilder()
+            .setKey(DATA_KEY)
+            .setShardingKey(SHARDING_KEY)
+            .setWorkToken(WORK_TOKEN)
+            .setMaxBytes(WindmillStateReader.MAX_CONTINUATION_KEY_BYTES)
+            .addMultimapsToFetch(
+                Windmill.TagMultimapFetchRequest.newBuilder()
+                    .setTag(STATE_KEY_1)
+                    .setStateFamily(STATE_FAMILY)
+                    .setFetchEntryNamesOnly(false)
+                    .addEntriesToFetch(
+                        Windmill.TagMultimapEntry.newBuilder()
+                            .setEntryName(STATE_MULTIMAP_KEY_1)
+                            .setFetchMaxBytes(WindmillStateReader.CONTINUATION_MAX_MULTIMAP_BYTES)
+                            .setRequestPosition(800)
+                            .build()));
+
+    Windmill.KeyedGetDataResponse.Builder response3 =
+        Windmill.KeyedGetDataResponse.newBuilder()
+            .setKey(DATA_KEY)
+            .addTagMultimaps(
+                Windmill.TagMultimapFetchResponse.newBuilder()
+                    .setTag(STATE_KEY_1)
+                    .setStateFamily(STATE_FAMILY)
+                    .addEntries(
+                        Windmill.TagMultimapEntry.newBuilder()
+                            .setEntryName(STATE_MULTIMAP_KEY_1)
+                            .addAllValues(Arrays.asList(intData(9), intData(10)))
+                            .setRequestPosition(800)));
+    Mockito.when(mockWindmill.getStateData(COMPUTATION, expectedRequest1.build()))
+        .thenReturn(response1.build());
+    Mockito.when(mockWindmill.getStateData(COMPUTATION, expectedRequest2.build()))
+        .thenReturn(response2.build());
+    Mockito.when(mockWindmill.getStateData(COMPUTATION, expectedRequest3.build()))
+        .thenReturn(response3.build());
+
+    Iterable<Integer> results = future.get();
+    Mockito.verify(mockWindmill).getStateData(COMPUTATION, expectedRequest1.build());
```

Comment:
any reason that this verify is in different spot than others?
Is there not a way to just verify everything you set up the when expectations for to avoid repeating?